roles/keystone/templates/etc/apache2/sites-available/keystone.conf

```diff
@@ -16,37 +16,49 @@ Listen {{ endpoints.keystone.port.backend_api }}
 
 {% macro setup_oidc() -%}
     OIDCClaimPrefix "OIDC_CLAIM_"
-    OIDCScope "{{ keystone.federation.sp.oidc.scope }}"
-    OIDCProviderIssuer {{ keystone.federation.sp.oidc.issuer }}
     OIDCClaimDelimiter ";"
-
+    OIDCMetadataDir /etc/apache2/openidc/metadata
     # oidc
     {% if not keystone.federation.sp.oidc.ssl_validate_server|bool -%}
     OIDCSSLValidateServer Off
     {% endif -%}
-    OIDCProviderUserInfoEndpoint {{ keystone.federation.sp.oidc.user_info_endpoint }}
-    OIDCProviderAuthorizationEndpoint {{ keystone.federation.sp.oidc.authorization_endpoint }}
-    OIDCProviderTokenEndpoint {{ keystone.federation.sp.oidc.token_endpoint }}
-    OIDCClientID {{ keystone.federation.sp.oidc.client_id }}
-    OIDCClientSecret {{ keystone.federation.sp.oidc.client_secret }}
     OIDCCryptoPassphrase {{ keystone.federation.sp.oidc.crypto_passphrase }}
     OIDCRedirectURI {{ keystone.federation.sp.oidc.redirect_uri }}
 
     # oauth
-    {% if not keystone.federation.sp.oidc.ssl_validate_server|bool -%}
+
```

Comment:
Would there be a way to handle additional parameters here. There could be other oauth directives that may be needed for specific providers.